api/v1alpha1/addon_types.go

```diff
@@ -90,6 +90,9 @@ type NutanixAddons struct {
 
 	// +kubebuilder:validation:Optional
 	COSI *NutanixCOSI `json:"cosi,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	KonnectorAgent *NutanixKonnectorAgent `json:"konnectorAgent,omitempty"`
```

Comment:
Is there a usecase where would not want to register a Cluster? Im wondering if this is too low level even giving the choice here? Can we always enable this?
My biggest concern is the `addons.KonnectorAgent` its really not an addon that would be extended for other providers. Maybe a better place for this would be in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/58197d650c8916c7273c3ba22bcb12bf0ea6cf62/api/v1alpha1/nutanix_clusterconfig_types.go#L18